CSS `content: url() / "alternative text" fail case (#21801)
mdn/content: files/en-us/web/css/content/index.md

/* <string> value */
content: "prefix";

/* list of content values */
content: "prefix" url("http://www.example.com/test.png");

> content: "prefix" url("http://www.example.com/test.png") "suffix" / "This is some alt text";

Author: It was not obvious to me from this document that you could have more than one thing in the content list. So I added this part, and the `<content-list>` in values below.

  - : Computes to `none` for the `::before` and `::after` pseudo-elements.
- {{cssxref("&lt;string&gt;")}}
  - : Specifies the "alt text" for the element. This value can be any number of text characters. Non-Latin characters must be encoded using their Unicode escape sequences: for example, `\000A9` represents the copyright symbol.
- `<content-list>`
  - : A list of anonymous inline boxes that will replace the content of the selected element (in the specified order).

>     This list can include strings, images, counters, and so on.

Author: This is what makes it clear you don't have to have just one string or image or whatever - you can have a list. This should be any of the following types. We could indent them below it if you like. I decided not to for now.

  - : These values are replaced by the appropriate string from the {{cssxref("quotes")}} property.
- `no-open-quote` | `no-close-quote`
  - : Introduces no content, but increments (decrements) the level of nesting for quotes.

> - `<content-list> / "Alternative text"`

Author: This is how I represent the alternative text - it has to be after the content list, so it can't just be on its own. Note in the next line I say "specified for an image (or list of ...). That's deliberate - I'm sure that this will most often be used for images.

#### CSS

The CSS to shown the image if present and otherwise the alternative text is shown below.

```css
a::before {

>   content: url("https://mozorg.cdn.mozilla.net/media/img/xfavicon.ico") /

Author: I changed this to `xfavicon.ico` to confirm the text was being displayed. Note however on Chrome this text is not being displayed - even though chrome support is indicated, and if this was not supported I would expect the case where the image is present to not work. 

1. What's going on? Chrome bug?
2. If we can get this to work, should we invert this permanently? I.e. is it more useful to show the fallback happening or the success case?

Author: Rachel Andrews thinks Chrome behaviour is probably valid. I'm seeking a second opinion in https://bugzilla.mozilla.org/show_bug.cgi?id=1281158#c7 because it doesn't make sense to me - if I use this syntax I really want either an image or text, and I don't care why the text might not be present.

>   content: url("https://mozorg.cdn.mozilla.net/media/img/xfavicon.ico") /

Reviewer: ```suggestion
  content: url("https://mozorg.cdn.mozilla.net/media/img/xfavicon.ico");
  content: url("https://mozorg.cdn.mozilla.net/media/img/xfavicon.ico") /
```

does this work as fallback: if the second is supported, it should be invalid. The previous line *is* valid, so should work. Should we include this (if it is correct)?

Author: Yes it is correct, but no I don't think we should include it. The short reason is that I'm going to revert this to just showing the "success" case.

The longer reason is that either we want this to fail to show the alt-text working OR we want it to succeed showing the image. I put in the xfavicon (which does not exist) to test the fail case. Unfortunately it doesn't fail "as it should" which I am still exploring in https://bugzilla.mozilla.org/show_bug.cgi?id=1281158#c7.

I still think showing the case where the fallback text is displayed would be more interesting here, but there is no easy way to demonstrate that unless I can find an URL for the image that exists, but is not valid for some kind of codec reason. 

The only reason the invalid icon is there is to see if I can force the fail case.

Unfortunately I can't know if this is 

Before we can decide how this looks I really need confirmation on whether or not chrome is broken - https://github.com/mdn/content/pull/21801/files#r1003858095

#### Result

The browser should display the icon before the link below, as the image is expected to be present.

{{EmbedLiveSample('Image_combined_with_text', '100%', 60)}}

Note that on a browser that does not support the alternative text syntax, the whole line is invalid.
In this case neither the image or text will be displayed!

> Depending on whether the image is _likely_ to be present you could address this issue by having CSS to add either the image or default text before the line with the both.

Reviewer: ```suggestion
Depending on whether the image is _likely_ to be present, you could address this issue by including CSS that adds the image or the default text before the line with them both.
```
